modules/monitoring-attaching-additional-labels-to-your-time-series-and-alerts.adoc

```diff
@@ -0,0 +1,58 @@
+// Module included in the following assemblies:
+//
+// * monitoring/cluster_monitoring/configuring-the-monitoring-stack.adoc
+
+[id="attaching-additional-labels-to-your-time-series-and-alerts_{context}"]
+= Attaching additional labels to your time series and alerts
+
+Using the external labels feature of Prometheus, you can attach additional custom labels to all time series and alerts leaving the Prometheus cluster.
+
+.Prerequisites
+
+* Make sure you have the `cluster-monitoring-config` ConfigMap object with the `data/config.yaml` section.
+
+.Procedure
+
+. Start editing the `cluster-monitoring-config` ConfigMap:
++
+----
+$ oc -n openshift-monitoring edit configmap cluster-monitoring-config
+----
+
+. Define a map of labels you want to add for every metric under `data/config.yaml`:
++
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      externalLabels:
+        <key>: <value>
+----
++
+Substitute `<key>: <value>` with a map of key-value pairs where `<key>` is a unique name of the new label and `<value>` is its value.
++
+[WARNING]
+====
+Do not use `prometheus` or `prometheus_replica` as key names, because they are reserved and would be overwritten.
+====
++
+For example, to add metadata about the region and environment to all time series and alerts, use:
++
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      externalLabels:
+        region: eu
+        environment: prod
+----
```

Comment:
Just wondering if there should be a final step of something like "Save the file to apply the changes."

Reply:
Good catch, I forgot it.